src/instructions/tlb.rs

```diff
@@ -97,3 +102,69 @@ pub unsafe fn flush_pcid(command: InvPicdCommand) {
         asm!("invpcid {0}, [{1}]", in(reg) kind, in(reg) &desc, options(nostack, preserves_flags));
     }
 }
+
+/// Invalidates TLB entry(s) with Broadcast.
+///
+/// # Safety
+///
+/// This function is unsafe as it requires CPUID.(EAX=8000_0008H, ECX=0H):EBX.INVLPGB
+/// to be 1 and count to be less than or equal to CPUID.(EAX=8000_0008H, ECX=0H):EDX[0..=15].
+#[inline]
+pub unsafe fn flush_broadcast<S>(
+    va_and_count: Option<(Page<S>, u16)>,
+    pcid: Option<Pcid>,
+    asid: Option<u16>,
+    include_global: bool,
+    final_translation_only: bool,
+    include_nested_translations: bool,
+) where
+    S: NotGiantPageSize,
```

Comment:
Do we want to use the builder pattern for this? There are an awful lot of parameters and this doesn't seem very future proof in case AMD adds more options.

Reply:
I agree that we want some sort of builder or [`#[non_exhaustive]` struct](https://doc.rust-lang.org/reference/attributes/type_system.html#the-non_exhaustive-attribute) to be used for this.

A builder pattern would also better reflect:
  - What reasonable defaults should be
  - What parameters are optional

Reply:
This can also be used to make some parameters unsafe.